ui/src/app/views/homepage/component.tsx

```diff
@@ -0,0 +1,394 @@
+                                  failedText=''
+                                  onClick={() => {}}/>
+          </div>
+        </div>
+      </div>
+    </React.Fragment>;
+  }
+}
+
+@Component({
+  selector: 'app-account-linking',
+  template: '<div #root></div>',
+})
+export class AccountLinkingComponent extends ReactWrapperBase {
+  @Input('eraCommonsLinked') eraCommonsLinked: AccountLinkingProps['eraCommonsLinked'];
+  @Input('trainingCompleted') trainingCompleted: AccountLinkingProps['trainingCompleted'];
+  constructor() {
+    super(AccountLinking,
+        ['eraCommonsLinked', 'trainingCompleted']);
+  }
+}
+
+
+@Component({
+  styleUrls: ['./component.css'],
+  templateUrl: './component.html',
+})
+
+export class HomepageComponent implements OnInit, OnDestroy {
+  private static pageId = 'homepage';
+  @ViewChild('myVideo') myVideo: any;
+  profile: Profile;
+  view: any[] = [180, 180];
+  numberOfTotalTasks = 4;
+  completedTasksName = 'Completed';
+  unfinishedTasksName = 'Unfinished';
+  open = false;
+  src = '';
+  spinnerValues = [
+    {
+      'name': this.completedTasksName,
+      'value': this.completedTasks
+    },
+    {
+      'name': this.unfinishedTasksName,
+      'value': this.numberOfTotalTasks - this.completedTasks
+    }
+  ];
+  billingProjectInitialized = false;
+  billingProjectQuery: NodeJS.Timer;
+  firstSignIn: Date;
+  firstVisit = true;
+  newPageVisit: PageVisit = { page: HomepageComponent.pageId};
+  footerLinks = [
+    {
+      title: 'Working Within Researcher Workbench',
+      links: ['Researcher Workbench Mission',
+        'User interface components',
+        'What to do when things go wrong',
+        'Contributing to the Workbench']
+    },
+    {
+      title: 'Workspace',
+      links: ['Workspace interface components',
+        'User interface components',
+        'Collaborating with other researchers',
+        'Sharing and Publishing Workspaces']
+    },
+    {
+      title: 'Working with Notebooks',
+      links: ['Notebook interface components',
+        'Notebooks and data',
+        'Collaborating with other researchers',
+        'Sharing and Publishing Notebooks']
+    }];
+  @ViewChild(BugReportComponent)
+  bugReportComponent: BugReportComponent;
+  quickTour: boolean;
+  accountsLinked: boolean;
+  // TODO RW-1184; defaulting to true
+  trainingCompleted = true;
+
+  constructor(
+    private profileService: ProfileService,
+    private profileStorageService: ProfileStorageService,
+    private route: ActivatedRoute,
+    private router: Router,
+  ) {
+    // create bound methods to use as callbacks
+    this.closeQuickTour = this.closeQuickTour.bind(this);
+  }
+
+  ngOnInit(): void {
+    this.profileService.getMe().subscribe(profile => {
+      if (profile.pageVisits) {
+        this.firstVisit = !profile.pageVisits.some(v =>
+        v.page === HomepageComponent.pageId);
+      }
+      if (environment.enableComplianceLockout) {
```

Comment:
This should be a combination of `enforceRegistered` and the new flag, most likely.